### ISSUE TITLE
Endianness detection

### ISSUE THREAD

**Issue by smimram** [savonet/ocaml-shine#3]
The endianness detection does not work here:
```
dune build @install
      ocamlc src/shine_stubs.o (exit 2)
(cd _build/default/src && /home/smimram/.opam/4.08.0/bin/ocamlc.opt -g -ccopt -g -o shine_stubs.o shine_stubs.c)
In file included from /usr/include/arpa/inet.h:22,
                 from shine_stubs.c:40:
shine_stubs.c:43:22: error: initializer element is not constant
 static int swapped = ntohs(1);
                      ^~~~~
```

**Comment by smimram** [comment]
Worse, you define `IS_BIGENDIAN`, but check for `BIGENDIAN` (not `IS_`...).

**Comment by smimram** [comment]
Fixed in 6047a4eed87a384a6444e4aad82e8045dae30958.
